raiden/api/python.py

```diff
@@ -1167,5 +1170,150 @@ def get_pending_transfers(
 
         return transfer_tasks_view(transfer_tasks, token_address, channel_id)
 
+    def set_total_udc_deposit(self, new_total_deposit: TokenAmount) -> TransactionHash:
+        """Set the `total_deposit` in the UserDeposit contract by sending an on-chain transaction.
+
+        Raises:
+            UserDepositNotConfigured: No UserDeposit is configured for the
+                Raiden node.
+            DepositMismatch: The new `total_deposit` is not higher than the
+                previous one.
+            DepositOverLimit: Either an overflow happened or the
+                `whole_balance_limit` of the UserDeposit contract is exceeded.
+            InsufficientFunds: Not enough tokens for the deposit.
+            RaidenRecoverableError: The transaction failed for any reason.
+
+        Returns: TransactionHash of the successfully mined transaction.
+        """
+        user_deposit = self.raiden.default_user_deposit
+
+        if user_deposit is None:
+            raise UserDepositNotConfigured("No UserDeposit contract is configured.")
+
+        chain_state = views.state_from_raiden(self.raiden)
+        confirmed_block_identifier = chain_state.block_hash
+
+        current_total_deposit = user_deposit.get_total_deposit(
+            address=self.address, block_identifier=confirmed_block_identifier
+        )
+        addendum = new_total_deposit - current_total_deposit
+
+        whole_balance = user_deposit.whole_balance(block_identifier=confirmed_block_identifier)
+        whole_balance_limit = user_deposit.whole_balance_limit(
+            block_identifier=confirmed_block_identifier
+        )
+
+        token_address = user_deposit.token_address(block_identifier=confirmed_block_identifier)
+        token = self.raiden.proxy_manager.token(
+            token_address, block_identifier=confirmed_block_identifier
+        )
+        balance = token.balance_of(
+            address=self.address, block_identifier=confirmed_block_identifier
+        )
+
+        if new_total_deposit <= current_total_deposit:
+            raise DepositMismatch("Total deposit did not increase.")
+
+        if whole_balance + addendum > UINT256_MAX:
+            raise DepositOverLimit("Deposit overflow.")
+
+        if whole_balance + addendum > whole_balance_limit:
+            msg = f"Deposit of {addendum} would have exceeded the UDC balance limit."
+            raise DepositOverLimit(msg)
+
+        if balance < addendum:
+            msg = f"Not enough balance to deposit. Available={balance} Needed={addendum}"
+            raise InsufficientFunds(msg)
```

Comment:
Should these checks be in the contract proxy, instead? Do we ever want to call the contract with any of these violated or unkown?

See also https://github.com/raiden-network/raiden/blob/develop/docs/proxies.rst#state-modifying-case

Reply:
They are in both places at the moment. This is quite redundant, but I tried to be consistent with the other API functions. The channel deposit function does the same for example.
One advantage I see is that we can show more specific error codes like this. For example a 402 if the RDN balance is too low.

The guide for using the proxies also says for a raised `BrokenPreconditionError`: 
> This means there is a mistake in the Raiden codebase, and a check must be added before calling the proxy.

And yes, it would be good to hear someone else's opinion on that.

Reply:
@ulope @palango Any opinions on this?

Reply:
I pointed this out repeatedly during our stand-up. Apparently nobody feels strong enough about this to comment. Feel free to proceed with the current approach.

Reply:
Alright then. Thanks for your help!